Create release-3.29.md (#3263)
sourcegraph/about: blogposts/2021/release-3.29.md

---
title: "Sourcegraph 3.29 release"

> publishDate: 2021-06-18T10:00-07:00

Reviewer: ```suggestion
publishDate: 2021-06-23T10:00-07:00
```

---
title: "Sourcegraph 3.29 release"
publishDate: 2021-06-18T10:00-07:00
description: "The Sourcegraph 3.29 release introduces improved search results ranking, makes it easier to create Code Insights, and includes support for bulk actions with Batch Changes."
tags: [blog, release]
slug: "release/3.29"

> published: false

Reviewer: ```suggestion
published: true
```

changelogItems:
  - description: "Code Insights queries can now run concurrently up to a limit set by the `insights.query.worker.concurrency` site config."
    url: https://github.com/sourcegraph/sourcegraph/pull/21219
    category: Code Insights
  - description: "Code Insights workers now support a rate limit for query execution and historical data frame analysis using the `insights.query.worker.rateLimit` and `insights.historical.worker.rateLimit` site configurations."
    url: https://github.com/sourcegraph/sourcegraph/pull/21533
    category: 
  - description: "The GraphQL `Site` `SettingsSubject` type now has an `allowSiteSettingsEdits` field to allow clients to determine whether the instance uses the `GLOBAL_SETTINGS_FILE` environment variable."
    url: https://github.com/sourcegraph/sourcegraph/pull/21827
    category: API

>   - description: "Code Insights creation UI now has auto-save logic and clear all fields functionality"

Reviewer: ```suggestion
  - description: "Code Insights creation UI now has autosave logic and clear all fields functionality."
```
- Making consistent with use of autosave elsewhere
- Let's standardise on using periods at the end of these

    url: https://github.com/sourcegraph/sourcegraph/pull/21744
    category: 

>   - description: Code Insights creation UI now has a suggestions support for the repository fields

Reviewer: ```suggestion
  - description: Code Insights creation UI now has suggestions support for the repository fields.
```

    url: https://github.com/sourcegraph/sourcegraph/pull/21699
    category: Repositories
  - description: A new bulk operation to retry many changesets at once has been added to Batch Changes.
    url: https://github.com/sourcegraph/sourcegraph/pull/21173
    category: Batch Changes
  - description: Added featured Sourcegraph extensions query to the GraphQL API, as well as a section in the extension registry to display featured extensions.
    url: https://github.com/sourcegraph/sourcegraph/pull/21665
    category: API

>   - description: "The search page now has a `create insight` button to create search-based insight based on your search query"

Reviewer: ```suggestion
  - description: "The search page now has a `create insight` button to create search-based insights based on your search query."
```

    url: https://github.com/sourcegraph/sourcegraph/pull/21364
    category: 
  - description: "Search streaming is now permanently enabled and `experimentalFeatures.searchStreaming` setting has been deprecated."
    url: https://github.com/sourcegraph/sourcegraph/pull/21522
    category: Search
  - description: "Sourcegraph will now refuse to start if there are unfinished [out-of-band-migrations](https://docs.sourcegraph.com/admin/migrations) that are deprecated in the current version. See the [upgrade documentation](https://docs.sourcegraph.com/admin/updates) for changes to the upgrade process."
    url: https://github.com/sourcegraph/sourcegraph/pull/20967
    category: Admin
# Fixed
  - description: "Stricter validation of structural search queries. The `type:` parameter is not supported for structural searches and returns an appropriate alert."
    url: https://github.com/sourcegraph/sourcegraph/pull/21487
    category: Search
  - description: Batch changeset specs that are not attached to changesets will no longer prematurely expire before the batch specs that they are associated with.
    url: https://github.com/sourcegraph/sourcegraph/pull/21678
    category: Batch Changes

>   - description: Code insights line chart no longer has a negative quadrant

Reviewer: ```suggestion
  - description: Code insights line chart no longer has a negative quadrant.
```

    category: Search
  - description: "The undocumented `index` filter was removed from search type-ahead suggestions."
    url: https://github.com/sourcegraph/sourcegraph/issues/18806
    category: Search
  - description: "Code host connection tokens aren't used for creating changesets anymore when the user is site admin and no credential has been specified."
    url: https://github.com/sourcegraph/sourcegraph/issues/16814
    category: Admin

---

Sourcegraph 3.29 is now available! For this release, we introduced search results ranking, made it easier to create Code Insights, and added support for bulk actions with Batch Changes.

## Search results ranking
We're making search results more relevant, starting with better prioritization of repos vs. code in your search results. Search results and suggestions are now ranked by descending star count on sourcegraph.com and customer instances (for repos with stars), making it easier to find specific repos. Start searching on sourcegraph.com to experience the new ranking.

Star count import from github.com/sourcegraph/ghdump is done. We now have 418321 repos with star data in the production repo table.

> Suggestions and search results are now ranked by descending star count on sourcegraph.com (and customer instances if repos there have stars). Try it out live!

Reviewer: I think this line was maybe meant to be deleted when the reworded paragraph above was inserted? Appears to be repetition

## New visual design for the Sourcegraph UI

> We are proud to bring you an entirely new visual design for the Sourcegraph UI. The new visual design was developed to help users find references, troubleshoot errors, gain insight, make changes on a massive scale, and read code. Check out the [blog post](https://about.sourcegraph.com/blog/introducing-sourcegraphs-new-ui/) to learn more.

Reviewer: ```suggestion
We are proud to bring you an entirely new visual design for the Sourcegraph UI. The new visual design was developed to help users find references, troubleshoot errors, gain insight, make changes on a massive scale, and read code. Check out our recent blog post to [learn more about the new Sourcegraph UI](/blog/introducing-sourcegraphs-new-ui/).
```
I try to use meaningful link text to make links more accessible to folks using screen readers

> ![refined search results design screenshot](https://sourcegraphstatic.com/blog/redesign/r_search_results.png)

Reviewer 2: Sure, looking at this now!

> ![refined search results design screenshot](https://sourcegraphstatic.com/blog/redesign/r_search_results.png)

Reviewer 2: @rebeccadee this is fixed. I have written up the larger problem and will also post to #marketing.

## Code Insights

### One-click Code Insights creation from your search

<p><video autoplay loop muted playsinline style="width:600px">
  <source src="https://sourcegraphstatic.com/blog/3.29/one_click_create_insight.mp4" type="video/mp4">
 </video></p>
 

>  You can now create code insights with 1 click, right from the search query view. Take your search query and turn it into a visualization of your results count in your code over time, to see if your usage of an API, function, or other target is growing or shrinking.

Reviewer: ```suggestion
 You can now create code insights with one click, right from the search query view. Take your search query and turn it into a visualization of your results count in your code over time, to see if your usage of an API, function, or other target is growing or shrinking.
```
https://about.sourcegraph.com/handbook/communication/content_guidelines/style_and_mechanics#numbers

### Improved creation forms for Code Insights

<p><video autoplay loop muted playsinline style="width:600px">
  <source src="https://sourcegraphstatic.com/blog/3.29/autosuggest_autosave_code_insights.mp4" type="video/mp4">
 </video></p>

 We now autosuggest and autocomplete the repository field in the correct format. The forms also now autosave your configuration, in case you navigate away before you hit "create." (You can clear your existing autosave if you want to start fresh, with the "clear all fields" button.)

## Batch Changes

### New bulk actions: retry and merge (experimental)

Last release we introduced [bulk actions](https://about.sourcegraph.com/blog/release/3.28/#Batch-changes), allowing users to comment on hundreds of changesets in a few clicks. We've added two more bulk actions in this release: retry errored changesets and merge.

> Merge is released as an experimental feature. It supports merge commit and squash merge on GitHub and GitLab, and merge commit on Bitbucket. For example, you can now filter all changesets that have passed checks and review, and merge them all in a few clicks. [Read more](https://docs.sourcegraph.com/batch_changes/how-tos/bulk_operations_on_changesets#supported-types-of-bulk-operations).

Reviewer: ```suggestion
Merge is released as an experimental feature. It supports merge commit and squash merge on GitHub and GitLab, and merge commit on Bitbucket. For example, you can now filter all changesets that have passed checks and review, and merge them all in a few clicks. [Read more about bulk operations](https://docs.sourcegraph.com/batch_changes/how-tos/bulk_operations_on_changesets#supported-types-of-bulk-operations).
```

---
title: "Sourcegraph 3.29 release"
publishDate: 2021-06-18T10:00-07:00
description: "The Sourcegraph 3.29 release introduces improved search results ranking, makes it easier to create Code Insights, and includes support for bulk actions with Batch Changes."
tags: [blog, release]
slug: "release/3.29"
published: false
changelogItems:
  - description: "Code Insights queries can now run concurrently up to a limit set by the `insights.query.worker.concurrency` site config."
    url: https://github.com/sourcegraph/sourcegraph/pull/21219
    category: Code Insights
  - description: "Code Insights workers now support a rate limit for query execution and historical data frame analysis using the `insights.query.worker.rateLimit` and `insights.historical.worker.rateLimit` site configurations."
    url: https://github.com/sourcegraph/sourcegraph/pull/21533
    category: 
  - description: "The GraphQL `Site` `SettingsSubject` type now has an `allowSiteSettingsEdits` field to allow clients to determine whether the instance uses the `GLOBAL_SETTINGS_FILE` environment variable."
    url: https://github.com/sourcegraph/sourcegraph/pull/21827
    category: API

>   - description: "Code Insights creation UI now has auto-save logic and clear all fields functionality"

Reviewer 3: I agree with Joel, and that anything mentioned in the highlights should have the short summary in the changelog below.

>   - description: "Code Insights creation UI now has auto-save logic and clear all fields functionality"

Author: @christinaforney @Joelkw - we can definitely keep this in the release post. 

Previously, we had the release post (blog) and the changelog. The guidance from Christina was that the release post included a curated list of updates and the changelog was comprehensive and included everything. 

@Joelkw - are you suggesting that the release post should still include every single update? If yes, then should we just link to the changelog?

>   - description: "Code Insights creation UI now has auto-save logic and clear all fields functionality"

Reviewer 2: @abschu I got similar guidance from Christina that I used for the 3.23 and 3.24 release posts and I think "curated list of updates" means what this post currently is without the suggested removals you commented: it contains a curated list of updates in the changelog section of the blog post (rather than the entire changelog from changelog.md). In addition to that "curated list" (again, in the changelog section of a release bog post), we pull out items we want to add images/gifs/videos of and add description. 

I don't think anyone would care strongly if you wanted to also link to the changelog.md itself (possibly an argument we shouldn't call anything in the blog post "changelog" if it's not complete), but we'd need a way to communicate/name the highlights we pulled out into the blog post list and why they're different.

>   - description: "Code Insights creation UI now has auto-save logic and clear all fields functionality"

Author: That sounds good - happy to leave these items in here.

---
title: "Sourcegraph 3.29 release"
publishDate: 2021-06-18T10:00-07:00
description: "The Sourcegraph 3.29 release introduces improved search results ranking, makes it easier to create Code Insights, and includes support for bulk actions with Batch Changes."
tags: [blog, release]
slug: "release/3.29"
published: false
changelogItems:
  - description: "Code Insights queries can now run concurrently up to a limit set by the `insights.query.worker.concurrency` site config."
    url: https://github.com/sourcegraph/sourcegraph/pull/21219
    category: Code Insights

>   - description: "Code Insights workers now support a rate limit for query execution and historical data frame analysis using the `insights.query.worker.rateLimit` and `insights.historical.worker.rateLimit` site configurations."

Reviewer 2: @abschu I'll leave the gameday decision to you but I would suggest keeping it because:
a) it's a setting that an end user / admin can set, so it's not just something happening behind the scenes
b) my guess is primarily admins (people who could make this setting update if they wanted) are reading the changelog closely so they know why/when to upgrade.

>   - description: "Code Insights workers now support a rate limit for query execution and historical data frame analysis using the `insights.query.worker.rateLimit` and `insights.historical.worker.rateLimit` site configurations."

Author: Let's keep - thank you!

    url: https://github.com/sourcegraph/sourcegraph/pull/21533
    category: 
  - description: "The GraphQL `Site` `SettingsSubject` type now has an `allowSiteSettingsEdits` field to allow clients to determine whether the instance uses the `GLOBAL_SETTINGS_FILE` environment variable."
    url: https://github.com/sourcegraph/sourcegraph/pull/21827
    category: API
  - description: "Code Insights creation UI now has auto-save logic and clear all fields functionality"
    url: https://github.com/sourcegraph/sourcegraph/pull/21744
    category: 
  - description: Code Insights creation UI now has a suggestions support for the repository fields
    url: https://github.com/sourcegraph/sourcegraph/pull/21699
    category: Repositories
  - description: A new bulk operation to retry many changesets at once has been added to Batch Changes.
    url: https://github.com/sourcegraph/sourcegraph/pull/21173
    category: Batch Changes

>   - description: "A `security_event_logs` database table has been added in support of upcoming security-related efforts."

Reviewer 4: It's data we want to log so we can troubleshoot any future potential security incident affecting Cloud.

I'd not suggest we include this.

    url: https://github.com/sourcegraph/sourcegraph/pull/21949
    category: 
  - description: Added featured Sourcegraph extensions query to the GraphQL API, as well as a section in the extension registry to display featured extensions.
    url: https://github.com/sourcegraph/sourcegraph/pull/21665
    category: API
  - description: "The search page now has a `create insight` button to create search-based insight based on your search query"
    url: https://github.com/sourcegraph/sourcegraph/pull/21943
    category: Search
  - description: Added support for Terraform syntax highlighting.
    url: https://github.com/sourcegraph/sourcegraph/pull/22040
    category: 
  - description: A new bulk operation to merge many changesets at once has been added to Batch Changes.
    url: https://github.com/sourcegraph/sourcegraph/pull/21959
    category: Batch Changes
  - description: Pings include aggregated usage for the Code Insights creation UI, organization visible insight count per insight type, and insight step size in days.
    url: https://github.com/sourcegraph/sourcegraph/pull/21671
    category: 

>   - description: "Search-based insight creation UI now supports `count:` filter in data series query input."

Reviewer: Should we move this directly beneath the related item?

    url: https://github.com/sourcegraph/sourcegraph/pull/22049
    category: Search
# Changed
  - description: SSH public keys generated to access code hosts with batch changes now include a comment indicating they originated from Sourcegraph.
    url: https://github.com/sourcegraph/sourcegraph/issues/20523
    category: Batch Changes
  - description: "The copy query button is now permanently enabled and `experimentalFeatures.copyQueryButton` setting has been deprecated."
    url: https://github.com/sourcegraph/sourcegraph/pull/21364
    category: 
  - description: "Search streaming is now permanently enabled and `experimentalFeatures.searchStreaming` setting has been deprecated."
    url: https://github.com/sourcegraph/sourcegraph/pull/21522
    category: Search

>   - description: Pings removes the collection of aggregate search filter usage counts and adds a smaller set of aggregate usage counts for query operators, predicates, and pattern counts.

Author: @christinaforney - I recommend cutting.

    url: https://github.com/sourcegraph/sourcegraph/pull/21320
    category: Search
  - description: "Sourcegraph will now refuse to start if there are unfinished [out-of-band-migrations](https://docs.sourcegraph.com/admin/migrations) that are deprecated in the current version. See the [upgrade documentation](https://docs.sourcegraph.com/admin/updates) for changes to the upgrade process."
    url: https://github.com/sourcegraph/sourcegraph/pull/20967
    category: Admin

>   - description: Code Insight pages now have new URLs

Author: @Joelkw - I recommend cutting

## Search results ranking
We're making search results more relevant, starting with better prioritization of repos vs. code in your search results. Search results and suggestions are now ranked by descending star count on sourcegraph.com and customer instances (for repos with stars), making it easier to find specific repos. Start searching on sourcegraph.com to experience the new ranking.

Star count import from github.com/sourcegraph/ghdump is done. We now have 418321 repos with star data in the production repo table.
Suggestions and search results are now ranked by descending star count on sourcegraph.com (and customer instances if repos there have stars). Try it out live!

## New visual design for the Sourcegraph UI
We are proud to bring you an entirely new visual design for the Sourcegraph UI. The new visual design was developed to help users find references, troubleshoot errors, gain insight, make changes on a massive scale, and read code. Check out the [blog post](https://about.sourcegraph.com/blog/introducing-sourcegraphs-new-ui/) to learn more.

> (Add screenshot)

Author: @rrhyne - it'd be great to feature the new visual design in the 3.29 release post. Can you please add a screenshot to this section?

Reviewer 5: Done!

---
title: "Sourcegraph 3.29 release"
publishDate: 2021-06-18T10:00-07:00
description: "The Sourcegraph 3.29 release includes."
tags: [blog, release]
slug: "release/3.29"
published: false
changelogItems:

---

Sourcegraph 3.29 is now available! For this release, we added

## Search results ranking

> Star count import from github.com/sourcegraph/ghdump is done. We now have 418321 repos with star data in the production repo table.

Author: We're making search ranking more intelligent, starting with better prioritization of repos vs. code in your search results. Search results and suggestions are now ranked by descending star count on sourcegraph.com and customer instances (for repos with stars), making it easier to find specific repos. Start searching on sourcegraph.com to experience the new ranking.
